### PR TITLE
Upgrade to ES61

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: sql/src/test/java/io/crate/metadata/doc/DocIndexMetaDataTest.java [hunk 1]
```diff
@@ -942,7 +943,7 @@ private DocIndexMetaData getDocIndexMetaDataFromStatement(String stmt) throws IO
             new IndexNameExpressionResolver(Settings.EMPTY)
         );
         DocSchemaInfo docSchemaInfo = new DocSchemaInfo(Schemas.DOC_SCHEMA_NAME, clusterService, functions, udfService, docTableInfoFactory);
-        CreateTableStatementAnalyzer analyzer = new CreateTableStatementAnalyzer(
+        Path homeDir = createTempDir();CreateTableStatementAnalyzer analyzer = new CreateTableStatementAnalyzer(
```
Comment (matriv) [inline sql/src/test/java/io/crate/metadata/doc/DocIndexMetaDataTest.java]: change line plz

File: sql/src/test/java/io/crate/lucene/LuceneQueryBuilderTest.java [hunk 2]
```diff
@@ -200,14 +204,14 @@ private MapperService createMapperService(IndexSettings indexSettings, IndexAnal
             indexSettings,
             indexAnalyzers,
             xContentRegistry(),
-            new SimilarityService(indexSettings, Collections.emptyMap()),
+            new SimilarityService(indexSettings, mock(ScriptService.class), Collections.emptyMap()),
```
Comment (matriv) [inline sql/src/test/java/io/crate/lucene/LuceneQueryBuilderTest.java]: also use defined variable `scriptService` here?

File: sql/src/main/java/io/crate/metadata/FulltextAnalyzerResolver.java [hunk 3]
```diff
@@ -196,26 +198,25 @@ public boolean hasBuiltInTokenFilter(String name) {
...
         }
     }
 
     public static Settings decodeSettings(String encodedSettings) throws IOException {
-        Map<String, String> loaded = new JsonSettingsLoader(false).load(encodedSettings);
-        return Settings.builder().put(loaded).build();
-
-
+        return Settings.builder().loadFromSource(encodedSettings, XContentType.JSON).build();
+        /*
```
Comment (matriv) [inline sql/src/main/java/io/crate/metadata/FulltextAnalyzerResolver.java]: remove commented code

File: enterprise/users/src/main/java/io/crate/auth/HostBasedAuthentication.java [hunk 4]
```diff
@@ -81,7 +82,12 @@ public HostBasedAuthentication(Settings settings, UserLookup userLookup) {
         Settings hbaSettings = AuthSettings.AUTH_HOST_BASED_CONFIG_SETTING.setting().get(settings);
         ImmutableSortedMap.Builder<String, Map<String, String>> hostBasedConf = ImmutableSortedMap.naturalOrder();
         for (Map.Entry<String, Settings> entry : hbaSettings.getAsGroups().entrySet()) {
-            hostBasedConf.put(entry.getKey(), entry.getValue().getAsMap());
+            Settings hbaEntry = entry.getValue();
```
Comment (matriv) [inline enterprise/users/src/main/java/io/crate/auth/HostBasedAuthentication.java]: Why is this needed?

Comment (mfussenegger) [reply enterprise/users/src/main/java/io/crate/auth/HostBasedAuthentication.java]: `getAsMap` is no longer available

File: sql/src/test/java/io/crate/analyze/CreateDropRepositoryAnalyzerTest.java [hunk 5]
```diff
@@ -21,6 +21,9 @@
 
 package io.crate.analyze;
 
+import com.google.common.base.Joiner;
```
Comment (matriv) [inline sql/src/test/java/io/crate/analyze/CreateDropRepositoryAnalyzerTest.java]: fix imports for this file please

File: sql/src/test/java/io/crate/execution/ddl/AlterTableOperationTest.java [hunk 6]
```diff
@@ -74,6 +74,7 @@ public void testOldSettingsAreArchivedOnPrepareIndexTemplateRequest() {
             .put(unsupportedSetting, false)
             .build();
         IndexTemplateMetaData indexTemplateMetaData = IndexTemplateMetaData.builder("t1")
+            .patterns(Collections.singletonList("*"))
```
Comment (matriv) [inline sql/src/test/java/io/crate/execution/ddl/AlterTableOperationTest.java]: is it mandatory now?

Comment (mfussenegger) [reply sql/src/test/java/io/crate/execution/ddl/AlterTableOperationTest.java]: yep, otherwise raises an error
